platform/orion/core/generic/vault.go

```diff
@@ -49,6 +49,14 @@ func (v *Vault) Status(txID string) (odriver.ValidationCode, error) {
 }
 
 func (v *Vault) DiscardTx(txid string) error {
+	vc, err := v.Vault.Status(txid)
+	if err != nil {
+		return errors.WithMessagef(err, "failed getting tx's status in state db [%s]", txid)
```

Comment:
I propose to use `errors.Wrapf(...` instead of `errors.WithMessagef` since `Wrap` also adds to the stack trace where we were wrapping, whereas `WithMessage` extends the error message.

Reply:
okay